File: Interpreted/ackermann-function/ack.py
import sys
sys.setrecursionlimit(10000)

def ack(M, N):
    if M == 0:
        return N + 1
    elif N == 0:
        return ack(M - 1, 1)
    else:
        return ack(M - 1, ack(M, N - 1))

print(ack(int(sys.argv[1]), int(sys.argv[2])))

File: Interpreted/ackermann-function/ack.ml
let rec a m n =
  if m=0 then (n+1) else
  if n=0 then (a (m-1) 1) else
  (a (m-1) (a m (n-1)))

let () =
  Printf.printf "%d\n" (a (int_of_string Sys.argv.(1)) (int_of_string Sys.argv.(2)))
